browser/views.py

```diff
@@ -220,24 +221,43 @@ def _handle_known_person_form(request, extra_form, person):
 
 
 @staff_member_required
-def edit_person(request, person_id):
-    person = get_object_or_404(Person, pk=person_id)
+def edit_person(request, person_id=None):
+    person = None
     if request.method == 'GET':
-        form = PersonForm(instance=person)
-        extra_form = KnownPersonForm(instance=person.authority, prefix='known')
+        if person_id:
+            person = get_object_or_404(Person, pk=person_id)
+            form = PersonForm(instance=person)
+            extra_form = KnownPersonForm(instance=person.authority, prefix='known')
+        else:
+            form = PersonForm()
+            extra_form = KnownPersonForm()
+
+    if request.method == 'POST':
+        if person_id:
+            person = get_object_or_404(Person, pk=person_id)
+            form = PersonForm(request.POST, instance=person)
+            extra_form = KnownPersonForm(request.POST, instance=person.authority, prefix='known')
+            if form.is_valid():
+                person = form.save()
+        else:
+            form = PersonForm(request.POST)
+            extra_form = KnownPersonForm(request.POST)
+            if form.is_valid():
+                person = Person.objects.create(
+                    changed_by=request.user,
+                    first_name=form.cleaned_data.get('first_name'),
+                    last_name=form.cleaned_data.get('last_name')
+                )
```

Comment:
why don't you save the form here as well?

Reply:
Instead of using <b>form.save</b>, I am using <b>objects.create</b> to create new entry in the database. I did so because <b>form.save</b> gives error as the <b>Person</b> model requires <b>changed_by</b> attribute but the <b>PersonForm</b> does not have one.

Reply:
that is not a good reason. If changed_by is required, then you need to provide it. There is a reason for that. We want to know who did what on the objects.

Reply:
I added the <b>changed_by</b> to <b>PersonForm</b> and made that field hidden in the template. Then, I replaced the above object creation code with <b>form.save()</b>